plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -60,16 +63,74 @@ export interface RouterOptions {
   reader: UrlReader;
   database: PluginDatabaseManager;
   catalogClient: CatalogApi;
+
   actions?: TemplateAction<any>[];
   taskWorkers?: number;
   taskBroker?: TaskBroker;
   additionalTemplateFilters?: Record<string, TemplateFilter>;
+  identity?: IdentityApi;
 }
 
 function isSupportedTemplate(entity: TemplateEntityV1beta3) {
   return entity.apiVersion === 'scaffolder.backstage.io/v1beta3';
 }
 
+function buildDefaultIdentityClient({
```

Comment:
Hold up, how come we don't use the default implementation here?

Reply:
This was re-introduced based on a conversation from @freben before.

The new DefaultItentityClient behaves slightly differently to the pre-existing scaffolder behaviour. Specifically if the token fails to parse, it will raise an error. The scaffolder did not raise an error in this case. As such we chose to allow it to behave as it did previously until someone explicitly passes an IdentityApi.

Reply:
Ah alright, probably worth it then as we'll have some workaround either way. Could you perhaps add comment with a bit of that reasoning? So that it's clearer in the future why this exists and to make it easier to get rid of when possible

Reply:
done :)